src/components/CmsContent.ts

```diff
@@ -1,17 +1,31 @@
+import isEqual from 'lodash/isEqual';
 import { CreateElement, Component as ComponentType, VNode } from 'vue';
-import { Vue, Component, Prop } from 'vue-property-decorator';
+import { Vue, Component, Prop, Watch } from 'vue-property-decorator';
 
 @Component({ name: 'cms-content' })
 export default class CmsContent extends Vue {
   @Prop(String) public html!: string;
   @Prop(Object) public extra!: {};
+  @Prop(Object) public renderContext!: {};
   @Prop(String) public zoneId!: string;
   @Prop({ default: '' }) public tag!: string;
 
   public context: object = {};
 
   private created(): void {
-    this.context = { ...this.extra };
+    this.context = {
+      ...this.extra,
+      ...this.renderContext,
+     };
+  }
+
+  @Watch('renderContext', { deep: true, immediate: true })
+  private updateContext(value?: object, oldValue?: object): void {
+    // Prevent deep watcher for extra from firing too often.
+    // See: https://github.com/vuejs/vue/issues/5776
+    if (oldValue !== undefined && !isEqual(value, oldValue)) {
+      Object.assign(this.context, this.renderContext);
+    }
```

Comment:
I'm probably missing something, but is there any reason `context` isn't a getter instead of a member variable? If it was, we wouldn't need to set it in `created()`, and we wouldn't need this watcher. Or?

Reply:
Is it because we don't want `extra` to trigger an update? Why wouldn't we want that?

Reply:
Context can be mutated within the CMS content itself via click and init handlers.